File: app/src/main/java/be/hogent/yasminedewinne/carwashapp/models/DTO/RegisterDTO.kt
package be.hogent.yasminedewinne.carwashapp.models.DTO

import be.hogent.yasminedewinne.carwashapp.models.domain.Adres
import com.squareup.moshi.Json

data class RegisterDTO(
    //var afbeelding: String?,

    var voornaam: String,

    var familienaam: String,

    var email: String,

    @Json(name = "TelefoonNr")
    var telNr: String,

    var adres: Adres,

    var wachtwoord: String,

    var wachtwoordHerhaling: String
)

File: API/CarwashAPI/DTO/Identity/RegisterDTO.cs
using System;
using System.ComponentModel.DataAnnotations;
using CarwashAPI.Models.Domain;

namespace CarwashAPI.DTO.Identity
{
    public class RegisterDTO
    {
        [Required]
        public string Voornaam { get; set; }

        [Required]
        public string Familienaam { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        [Phone]
        public string TelefoonNr { get; set; }

        [Required]
        public Adres Adres { get; set; }

        [Required]
        [StringLength(maximumLength: 50, MinimumLength = 6)]
        public string Wachtwoord { get; set; }

        [Required]
        [Compare(nameof(Wachtwoord))]
        public string WachtwoordHerhaling { get; set; }
    }
}
